src/context-selection/section-filter-selector-bar-item/section-filter-selector-bar-item.js

```diff
@@ -34,10 +42,14 @@ export default function SectionFilterSelectorBarItem() {
             label: displayName,
         })) || []
 
-    const selectableOptions = [
-        { value: '', label: i18n.t('Show all sections') },
-        ...sectionOptions,
-    ]
+    const selectableOptions = sectionOptions.concat({
+        value: undefined,
+        label: i18n.t('Show all sections'),
+    })
+
+    if (dataSetSectionsInfo.data?.renderAsTabs) {
+        selectableOptions.shift()
+    }
```

Comment:
```suggestion
    const selectableOptions = sectionOptions.concat(
        !dataSetSectionsInfo.data?.renderAsTabs
            ? { value: undefined, label: i18n.t('Show all sections') }
            : undefined
    )
```

Reply:
I actually had it reversed at first - and removed the option by using `unshift`. I also thought about justing a let and concating if `renderAsTabs=false`. 
Anyhow, I do not like concatenating `undefined` - then the options will have an undefined element in the array, and depending on how the child-component that consumes this handles that, you can easily get errors. 

If however you are against mutating the array, which don't really matter here since we are creating a new one everytime either way - I could do it like this

```js
const selectableOptions = dataSetSectionInfo.data?.renderAsTabs ? sectionOptions : sectionOptions.concat({ value: undefined, label: i18n.t('Show all sections') })
```

Reply:
Ah, I was under the impression concat won't append anything when the value is undefined, which doesn't make any sense of course.. Nevermind my comment!